lib/src/phoenix_presence.dart

```diff
@@ -0,0 +1,175 @@
+import 'dart:convert';
+import 'package:phoenix_wings/src/phoenix_channel.dart';
+
+class PhoenixPresence {
+  PhoenixChannel channel;
+  Map<String, dynamic> opts;
+  PresenceEvents events;
+  Map<String, Map<String, dynamic>> state;
+  List pendingDiffs = [];
+  String joinRef;
+  _PresenceCallers caller;
+
+  static void _noop(key, currentPresence, newPresence) {
+  }
+
+  PhoenixPresence(this.channel, {this.opts}) {
+    opts ??= {};
+    events = opts.containsKey('events') ? opts['events'] : PresenceEvents(PhoenixPresenceEvents.presenceState, PhoenixPresenceEvents.presenceDiff);
+    state = {};
+    joinRef = null;
+    caller = _PresenceCallers(_noop, _noop, () {});
```

Comment:
Maybe use named parameters for clarity

```suggestion
        caller = _PresenceCallers(onJoin: _noop, onLeave: _noop, onSync: () {});
```